ui/app/routes/vault/cluster/secrets/backend/configuration/index.js

```diff
@@ -5,28 +5,132 @@
+import { allEngines } from 'vault/helpers/mountable-secret-engines';
+import { hash } from 'rsvp';
+/**
+ * This route is responsible for fetching all configuration model(s).
+ * This includes the mount-configuration model attached to the secret-engine model via a belongsTo relationship.
+ * As well as any additional configuration models if the engine is a configurable engine.
+ */
 
 export default class SecretsBackendConfigurationRoute extends Route {
   @service store;
 
   async model() {
-    const backend = this.modelFor('vault.cluster.secrets.backend');
-    if (backend.isV2KV) {
+    const secretEngineModel = this.modelFor('vault.cluster.secrets.backend');
+    if (secretEngineModel.isV2KV) {
       const canRead = await this.store
-        .findRecord('capabilities', `${backend.id}/config`)
+        .findRecord('capabilities', `${secretEngineModel.id}/config`)
         .then((response) => response.canRead);
       // only set these config params if they can read the config endpoint.
       if (canRead) {
         // design wants specific default to show that can't be set in the model
-        backend.casRequired = backend.casRequired ? backend.casRequired : 'False';
-        backend.deleteVersionAfter = backend.deleteVersionAfter ? backend.deleteVersionAfter : 'Never delete';
+        secretEngineModel.casRequired = secretEngineModel.casRequired
+          ? secretEngineModel.casRequired
+          : 'False';
+        secretEngineModel.deleteVersionAfter = secretEngineModel.deleteVersionAfter
+          ? secretEngineModel.deleteVersionAfter
+          : 'Never delete';
       } else {
         // remove the default values from the model if they don't have read access otherwise it will display the defaults even if they've been set (because they error on returning config data)
-        backend.set('casRequired', null);
-        backend.set('deleteVersionAfter', null);
-        backend.set('maxVersions', null);
+        secretEngineModel.set('casRequired', null);
+        secretEngineModel.set('deleteVersionAfter', null);
+        secretEngineModel.set('maxVersions', null);
+      }
+    }
+    // If the engine is configurable fetch the config model(s) for the engine and return it alongside the model
+    if (CONFIGURABLE_SECRET_ENGINES.includes(secretEngineModel.type)) {
+      let configModels = await this.fetchConfig(secretEngineModel.type, secretEngineModel.id);
+      configModels = this.standardizeConfigModels(configModels);
+
+      return hash({
+        secretEngineModel,
+        ...configModels,
+      });
+    }
+    return hash({ secretEngineModel });
+  }
+
+  standardizeConfigModels(configModels) {
+    // standardize the configModels to an array so that the component can handle it correctly
+    Array.isArray(configModels) ? configModels : (configModels = [configModels]);
+    // make sure no items in the array are null or undefined
+    configModels.forEach((configModel) => {
```

Comment:
this would be better as a filter! 
```suggestion
    return configModels.filter((configModel) => !!configModel)
```